app/controllers/submissions_controller.rb

```diff
@@ -971,13 +971,17 @@ def notebook_to_html(file_contents, unique_path, type)
           '--template', 'markus-html-template'
         ]
       end
-      file_contents = JSON.parse(file_contents)
-      if file_contents['metadata'].key?('widgets')
-        file_contents['metadata'].delete('widgets')
+      if !SubmissionFile.is_binary?(file_contents)
```

Comment:
`rails_performance` is only logging unhandled exceptions. Rather than doing an `is_binary?` check here (which is a bit inefficient), you can instead wrap the `JSON.parse` call into a separate begin-rescue block, and rescue the `JSON::ParserError` to display the error message.

Reply:
Thank you @david-yz-liu for these great comments.